core/src/main/java/com/graphhopper/util/XFirstSearch.java

```diff
@@ -18,18 +18,18 @@
 package com.graphhopper.util;
 
 import com.graphhopper.coll.GHBitSet;
-import com.graphhopper.coll.GHBitSetImpl;
 
 /**
  * This abstract class defines commonalities for BFS and DFS
- * <p>
  *
  * @author Jan Sölter
  */
 public abstract class XFirstSearch {
-    protected GHBitSet createBitSet() {
-        return new GHBitSetImpl();
-    }
+    /**
+     * Pick the BitSet implementation wisely. Use GHBitSetImpl only if we are sure you visit a large portion of the graph.
+     * And if you choose GHTBitSet the initial capacity can be also important for performance.
+     */
+    protected abstract GHBitSet createBitSet();
```

Comment:
Ah this again :) Good idea to make this abstract so one has to pick the bit set explicitly.